src/main.rs

```diff
@@ -67,15 +70,46 @@ fn main() -> Result<(), error::Error> {
 
     let warnings_caused_by_diff =
         intersections::get_lints_from_diff(&clippy_lints, &diff_sections, opts.verbose);
-    if warnings_caused_by_diff.is_empty() {
+
+    return_warnings(&warnings_caused_by_diff, opts.without_error)
+}
+
+fn return_warnings(lints: &[clippy::Lint], without_error: bool) -> Result<(), error::Error> {
+    if lints.is_empty() {
         println!("No warnings raised by clippy::pedantic in your diff, you're good to go!");
         Ok(())
     } else {
-        display_warnings(&warnings_caused_by_diff);
-        println!(
-            "Clippy::pedantic found {} warnings",
-            warnings_caused_by_diff.len()
-        );
+        if without_error {
+            display_warnings(&lints);
```

Comment:
It seems we call display_warning(&lints) whether without_error is true or false, maybe we can call it just once, before the if statement ?

Reply:
That's true, somehow missed that..